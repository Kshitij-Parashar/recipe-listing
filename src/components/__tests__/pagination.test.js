import React from 'react';
import { mount } from 'enzyme';
import Pagination from '../pagination/pagination';
import '../../setupTests';

it('should render correctly <Pagination>', () => {
  const fakeProps = {
    postsPerPage: 5,
    totalPosts: 10,
    paginate: () => {},
    currentPage: 1,
  };
  const wrapper = mount(<Pagination {...fakeProps}/>);
  expect(wrapper).toMatchSnapshot();
});

it('should change page on action prev and next on prev and next button click', () => {
  const fakeProps = {
    postsPerPage: 5,
    totalPosts: 10,
    paginate: () => {},
    currentPage: 1,
  };

  const element = mount(
    <Pagination {...fakeProps} />
  );
  expect(element.find('#nextPage')).toHaveLength(1);
  expect(element.find('#nextPage')).toHaveLength(1);
  const nextPage = element.find('#nextPage');
  const prevPage = element.find('#prevPage');
  expect(element.find('#currentPage')).toHaveLength(1);
  nextPage.simulate('click')
  expect(element.find('#currentPage')).toHaveLength(1);
  expect(element.find('#currentPage').filterWhere((item) => {
    return item.prop('value') === 2;
  })).toHaveLength(1);
  prevPage.simulate('click');
  expect(element.find('#currentPage').filterWhere((item) => {
    return item.prop('value') === 1;
  })).toHaveLength(1);
});
