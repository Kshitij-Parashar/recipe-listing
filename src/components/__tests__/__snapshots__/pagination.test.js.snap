// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly <Pagination> 1`] = `
<Pagination
  currentPage={1}
  paginate={[Function]}
  postsPerPage={5}
  totalPosts={10}
>
  <input
    className="hide"
    id="currentPage"
    onChange={[Function]}
    value={1}
  />
  <div
    className="pagination"
  >
    <button
      className="previous"
      id="prevPage"
      onClick={[Function]}
    >
      « Prev
    </button>
    <button
      className="next"
      id="nextPage"
      onClick={[Function]}
    >
      Next »
    </button>
  </div>
</Pagination>
`;
